src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,70 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final RegistrationServiceImpl registrationService
+            = new RegistrationServiceImpl();
+    private static final StorageDaoImpl storageDaoImpl = new StorageDaoImpl();
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User("alice@gmail.com", "AliCe12345", 25);
+    }
+
+    @Test
+    void register_loginCheck_isNotOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageNull_isNotOk() {
+        user.setAge(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageUnder_isNotOk() {
+        user.setAge(13);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordNull_isNotOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordLength_isNotOk() {
+        user.setPassword("alice");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userRegistration_isNotOk() {
+        registrationService.register(user);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userNull_isNotOk() {
+        user = null;
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_loginCheckDuplicate_isOk() {
```

Comment:
we should register some valid user, and then try to register some new user with the same login, so we get an exception

Reply:
@MaliukDaria Thank you, already fixed ✓
I think, it's should be right